src/Controller/CancelController.php

```diff
@@ -15,36 +18,30 @@
  * limitations under the License.
  */
 
-namespace App\Controller;
+namespace Surfnet\Tiqr\Controller;
 
-use App\Exception\NoActiveAuthenrequestException;
 use Psr\Log\LoggerInterface;
 use Surfnet\GsspBundle\Service\AuthenticationService;
 use Surfnet\GsspBundle\Service\RegistrationService;
+use Surfnet\Tiqr\Exception\NoActiveAuthenrequestException;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
 class CancelController extends AbstractController
 {
-    private $authenticationService;
-    private $registrationService;
-    private $logger;
-
     public function __construct(
-        AuthenticationService $authenticationService,
-        RegistrationService $registrationService,
-        LoggerInterface $logger
+        private readonly AuthenticationService $authenticationService,
```

Comment:
Not sure, but if all the constructor params are readonly, you could use readonly class (althought that sometimes is hard(er) to mock)

Reply:
Yup that's why I'm reluctant in changing classes to be read-only. Leaving this for now. If/when phpunit makes this trivial. I'm open for suggestions.